test/brsTypes/components/RoSGNode.test.js

```diff
@@ -578,4 +578,130 @@ describe("RoSGNode", () => {
             });
         });
     });
+
+    describe("ifSGNodeDict", () => {
+        let interpreter, node;
+
+        beforeEach(() => {
+            interpreter = new Interpreter();
+            node = new RoSGNode([{ name: new BrsString("id"), value: new BrsString("root") }]);
+            child1 = new RoSGNode([{ name: new BrsString("id"), value: new BrsString("child1") }]);
+            child2 = new RoSGNode([{ name: new BrsString("id"), value: new BrsString("child2") }]);
+            child3 = new RoSGNode([{ name: new BrsString("id"), value: new BrsString("child3") }]);
+            child4 = new RoSGNode([{ name: new BrsString("id"), value: new BrsString("child4") }]);
+        });
+
+        describe("findnode", () => {
+            it("returns invalid if no node is found", () => {
+                let findNode = node.getMethod("findnode");
+                expect(findNode).toBeTruthy();
+
+                let invalidNode = findNode.call(interpreter, new BrsString("someRandomId"));
+                expect(invalidNode).toEqual(BrsInvalid.Instance);
+            });
+
+            it("finds itself", () => {
+                let findNode = node.getMethod("findnode");
+                expect(findNode).toBeTruthy();
+
+                let result = findNode.call(interpreter, new BrsString("root"));
+                expect(result).toBeTruthy();
+                expect(result).toBe(node);
+            });
+
+            it("finds a node in its direct children", () => {
+                let appendChild = node.getMethod("appendchild");
+                let children = [child1, child2, child3, child4];
+                for (let child of children) {
+                    appendChild.call(interpreter, child);
+                }
+
+                let findNode = node.getMethod("findnode");
```

Comment:
Don't think this check is needed here and below since you already did it in the first test case